base/float16.jl

```diff
@@ -160,7 +161,7 @@ end
 
 ldexp(a::Float16, b::Integer) = Float16(ldexp(Float32(a), b))
 
-^(x::Float16, y::Integer) = Float16(Float32(x)^y)
+^(x::Float16, y::Float16) = Float16(Float32(x)^Float32(y))
```

Comment:
This change is a mistake.

`^(x::Float16, y::Integer) = Float16(Float32(x)^y)` was correct.